src/ImageSharp/Formats/Tiff/TiffEncoderCore.cs

```diff
@@ -105,21 +125,33 @@ public void Encode<TPixel>(Image<TPixel> image, Stream stream, CancellationToken
 
             this.configuration = image.GetConfiguration();
 
-            TiffPhotometricInterpretation rootFramePhotometricInterpretation = GetRootFramePhotometricInterpretation(image);
-            TiffPhotometricInterpretation photometricInterpretation = this.PhotometricInterpretation == TiffPhotometricInterpretation.PaletteColor
-                ? TiffPhotometricInterpretation.PaletteColor
-                : rootFramePhotometricInterpretation;
-
             ImageFrameMetadata rootFrameMetaData = image.Frames.RootFrame.Metadata;
             TiffFrameMetadata rootFrameTiffMetaData = rootFrameMetaData.GetTiffMetadata();
-            TiffBitsPerPixel? rootFrameBitsPerPixel = rootFrameTiffMetaData.BitsPerPixel;
 
-            // If the user has not chosen a predictor or compression, set the values from the decoded image, if present.
-            this.HorizontalPredictor ??= rootFrameTiffMetaData.Predictor;
-            this.CompressionType ??= rootFrameTiffMetaData.Compression;
+            // Determine the correct values to encode with.
+            // EncoderOptions > Metadata > Default.
+            TiffBitsPerPixel? bitsPerPixel = this.BitsPerPixel ?? rootFrameTiffMetaData.BitsPerPixel;
+
+            TiffPhotometricInterpretation? photometricInterpretation = this.PhotometricInterpretation ?? rootFrameTiffMetaData.PhotometricInterpretation;
+
+            TiffPredictor predictor =
+                this.HorizontalPredictor
+                ?? rootFrameTiffMetaData.Predictor
+                ?? DefaultPredictor;
 
-            this.SetBitsPerPixel(rootFrameBitsPerPixel, image.PixelType.BitsPerPixel, photometricInterpretation);
-            this.SetPhotometricInterpretation(photometricInterpretation);
+            TiffCompression compression =
+                this.CompressionType
+                ?? rootFrameTiffMetaData.Compression
+                ?? DefaultCompression;
+
+            // Make sure, the bits per pixel and PhotometricInterpretation have values which makes sense in combination with the other chosen values.
+            bitsPerPixel = this.SanitizeBitsPerPixel(bitsPerPixel, image.PixelType.BitsPerPixel, photometricInterpretation, compression);
```

Comment:
Ok.... So.... I still find this part confusing. 

Are we taking `Compression` as the source of truth here?

`SantizeBitsPerPixel` is updating the `BitsPerPixel` value based upon values for `PhotometricInterpretation` which is then immediately updated in `SanitizePhotometricInterpretation` based upon the new `BitsPerPixel` value. It's all a bit chicken and egg. 

It seems to be that we have to treat one of those values as a source of truth. I would normally choose `BitsPerPixel` as that source.